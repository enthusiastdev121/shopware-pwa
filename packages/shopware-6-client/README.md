# API client for Shopware 6

Compatibility with Shopware 6:  
[![shopware-ver](https://img.shields.io/badge/version-6.1.1-orange)](https://github.com/shopware/platform/releases/tag/v6.1.1)

## Installing

Using yarn:

```bash
$ yarn add @shopware-pwa/shopware-6-client
```


## Usage example

A simple example of how to use API client to connect with your Shopware instance.

In main project file setup connection informations.
```js
import {setup} from "@shopware-pwa/shopware-6-client"

setup({
  endpoint: 'https://address-to-my-shopware-instance.com',
  accessToken: 'myaccesstoken'
})
```

And then anywhere in your project, you can use API Client methods:

```js
import { getCategories } from "@shopware-pwa/shopware-6-client"

// later in component

async mounted() {
  this.categories = await getCategories();
}
```