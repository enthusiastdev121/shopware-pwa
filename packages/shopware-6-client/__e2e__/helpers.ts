/**
 * Changes deeply specific property. Can be used in snapshot tests to mock changing value.
 */
export function deepChangeProperty(
  obj: any,
  property: string,
  value: any = "mockedValue"
) {
  deepChangeProperties(obj, [property, value]);
}

export function escapeQuotes(value: string) {
  return value.split('"').join("'");
}

export function deepEscapeProperties(obj: any, properties: string[]) {
  for (var prop in obj) {
    if (properties.includes(prop)) {
      if (typeof obj[prop] == "string") {
        obj[prop] = escapeQuotes(obj[prop]);
      }
    } else if (obj[prop] && obj[prop] === Object(obj[prop])) {
      deepEscapeProperties(obj[prop], properties);
    }
  }
}

export function deepChangeProperties(
  obj: any,
  properties: string[],
  value: any = "mockedValue"
) {
  for (var prop in obj) {
    if (properties.includes(prop) && Array.isArray(obj[prop])) {
      obj[prop] = [value];
    } else if (properties.includes(prop)) {
      obj[prop] = value;
    } else if (obj[prop] && obj[prop] === Object(obj[prop])) {
      deepChangeProperties(obj[prop], properties, value);
    }
  }
}
