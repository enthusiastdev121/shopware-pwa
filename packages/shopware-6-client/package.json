{
  "name": "@shopware-pwa/shopware-6-client",
  "version": "0.1.0-alpha.7",
  "description": "Rest API client for Shopware 6.",
  "main": "dist/shopware-6-client.cjs.js",
  "module": "dist/shopware-6-client.esm-bundler.js",
  "files": [
    "dist"
  ],
  "types": "dist/shopware-6-client.d.ts",
  "unpkg": "dist/shopware-6-client.global.js",
  "sideEffects": false,
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DivanteLtd/shopware-pwa"
  },
  "buildOptions": {
    "name": "Shopware6Client",
    "formats": [
      "esm-bundler",
      "cjs"
    ]
  },
  "peerDependencies": {
    "axios": "^0.19.2",
    "https": "^1.0.0"
  },
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  }
}
