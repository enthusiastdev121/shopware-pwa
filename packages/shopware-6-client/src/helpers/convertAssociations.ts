import {
  Association,
  ShopwareAssociation,
} from "@shopware-pwa/commons/interfaces/search/Association";

export function convertAssociations(
  associations: Association[] = []
): ShopwareAssociation | undefined {
  if (!Array.isArray(associations) || !associations.length) return;
  let shopwareAssociations: ShopwareAssociation = {};
  associations.forEach((association) => {
    shopwareAssociations[association.name] = association.associations
      ? {
          associations: convertAssociations(association.associations),
        }
      : {};
  });
  return shopwareAssociations;
}
