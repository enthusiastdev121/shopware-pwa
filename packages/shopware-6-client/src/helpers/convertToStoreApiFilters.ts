import {
  SearchFilter,
  SearchFilterType,
} from "@shopware-pwa/commons/interfaces/search/SearchFilter";

const concatIds = (ids: string[]) => ids.join("|");
const isFilterForProperty = (property: string, filter: any) =>
  filter.hasOwnProperty("field") && filter.field === property;

interface StoreApiListingFilters {
  manufacturer?: string;
  properties?: string;
  rating?: number; // to be handled later on
  "shipping-free"?: boolean; // to be handled later on
  "min-price"?: number; // to be handled later on
  "max-price"?: number; // to be handled later on
}

interface GenericFilter extends SearchFilter {
  value?: string[];
  queries?: GenericFilter[];
}

export function convertToStoreApiFilters(
  filters: Array<GenericFilter>
): StoreApiListingFilters {
  let params: StoreApiListingFilters = {};

  if (!filters || !filters.length) {
    return {};
  }

  for (const filter of filters) {
    if (isFilterForProperty("manufacturerId", filter) && filter.value) {
      params.manufacturer = concatIds(filter.value);
    }
    if (filter.type === SearchFilterType.MULTI && filter.queries) {
      for (const subFilter of filter.queries) {
        /* istanbul ignore else */
        if (isFilterForProperty("propertyIds", subFilter)) {
          params.properties = concatIds(subFilter.value as any);
        }
      }
    }
  }

  return params;
}
