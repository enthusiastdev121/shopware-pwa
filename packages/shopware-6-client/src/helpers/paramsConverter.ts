import { SearchCriteria } from "../interfaces/search/SearchCriteria";

export interface Params {
  page?: number;
  limit?: number;
  sort?: string;
  filter?: any;
}

/**
 * @description Combines parameters into one object
 */
export const getParams = (searchCriteria?: SearchCriteria): Params | null => {
  let params = {};

  if (!searchCriteria) return params;

  if (searchCriteria.pagination) {
    params = Object.assign(params, searchCriteria.pagination);
  }

  // if (sort) {
  //   params = Object.assign(params, sort);
  // }

  if (searchCriteria.filters) {
    params = Object.assign(params, searchCriteria.filters);
  }

  return params;
};
