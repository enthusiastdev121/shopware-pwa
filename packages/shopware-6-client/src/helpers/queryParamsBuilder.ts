import queryString from "query-string";

const ARRAY_FORMAT = "separator";
const ARRAY_FORMAT_SEPARATOR = "|";
const SKIP_NULL = true;
const SORT = false;

export const getQueryString = (params: any) =>
  typeof params === "string"
    ? params
    : queryString.stringify(params, {
        arrayFormat: ARRAY_FORMAT,
        arrayFormatSeparator: ARRAY_FORMAT_SEPARATOR,
        skipNull: SKIP_NULL,
        sort: SORT,
      });
