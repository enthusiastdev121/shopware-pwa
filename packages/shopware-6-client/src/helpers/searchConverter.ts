import { SearchCriteria } from "@shopware-pwa/commons/interfaces/search/SearchCriteria";
import {
  NotFilter,
  MultiFilter,
  RangeFilter,
  EqualsFilter,
  EqualsAnyFilter,
  SearchFilterType,
} from "@shopware-pwa/commons/interfaces/search/SearchFilter";
import { convertAssociations } from "./convertAssociations";
import { PaginationLimit } from "@shopware-pwa/commons/interfaces/search/Pagination";
import { config } from "@shopware-pwa/shopware-6-client";
import { ShopwareAssociation } from "@shopware-pwa/commons/interfaces/search/Association";
import { Grouping } from "@shopware-pwa/commons/interfaces/search/Grouping";

/**
 * @alpha
 */
export interface ShopwareParams {
  page?: number;
  limit?: number;
  sort?: string;
  term?: string;
  filter?: (
    | NotFilter
    | MultiFilter
    | EqualsFilter
    | EqualsAnyFilter
    | RangeFilter
  )[];
  associations?: ShopwareAssociation;
  grouping?: Grouping;
}

export const convertSearchCriteria = (
  searchCriteria?: SearchCriteria
): ShopwareParams => {
  let params: ShopwareParams = {};

  if (!searchCriteria) return params;
  const { filters, sort, pagination, configuration, term } = searchCriteria;

  if (pagination) {
    const { limit, page } = pagination;
    if (limit && Object.values(PaginationLimit).includes(limit))
      params.limit = limit;
    if (page) {
      params.page = page;
      if (!params.limit) params.limit = config.defaultPaginationLimit;
    }
  }

  if (sort) {
    let prefix = sort.desc ? "-" : "";
    params.sort = `${prefix}${sort.field}`;
  }

  if (filters && filters.length) {
    params.filter = filters;
  }

  if (configuration && configuration.associations) {
    params.associations = convertAssociations(configuration.associations);
  }

  if (configuration?.grouping) {
    params.grouping = configuration.grouping;
  }

  // fulltext query (works for every entity so can be global)
  if (term) {
    params.term = term;
  }

  // add extra grouping option and additional filter to prevent displaying redundand products.
  if (!configuration || (configuration && !configuration.displayParents)) {
    if (!Array.isArray(params.filter)) {
      params.filter = [];
    }
    params.grouping = {
      // prevent displaying parent instances of the product
      field: "displayGroup",
    };
    params.filter.push({
      type: SearchFilterType.NOT,
      queries: [
        {
          type: SearchFilterType.EQUALS,
          field: "displayGroup",
          value: null,
        },
      ],
    });
  }

  return params;
};
