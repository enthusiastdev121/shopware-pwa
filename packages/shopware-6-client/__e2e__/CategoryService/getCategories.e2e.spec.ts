import { getCategories } from "@shopware-pwa/shopware-6-client";
import { deepChangeProperties } from "../helpers";

describe("shopware-6-client - E2E - CategoryService - getCategories", () => {
  it("should fetch all categories that match search criteria", async () => {
    const result = await getCategories();
    deepChangeProperties(result, [
      "childCount",
      "afterCategoryId",
      "updatedAt",
      "total",
    ]);
    expect(result).toMatchSnapshot();
  });

  it("should returns error when cannot find categories with given criteria", async () => {
    try {
      await getCategories({ sort: { field: "qwa" } });
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
