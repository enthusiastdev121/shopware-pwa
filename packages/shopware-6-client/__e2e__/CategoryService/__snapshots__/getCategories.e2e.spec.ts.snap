// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CategoryService - getCategories should fetch all categories that match search criteria 1`] = `
Object {
  "aggregations": Array [],
  "data": Array [
    Object {
      "_uniqueIdentifier": "3f637f17cd9f4891a2d7625d19fb37c9",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": null,
      "autoIncrement": 1,
      "breadcrumb": Array [
        "Catalogue #1",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "3398fdb2898145f2b89915161b14738b",
      "createdAt": "2019-10-08T09:41:16+00:00",
      "customFields": null,
      "description": null,
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "3f637f17cd9f4891a2d7625d19fb37c9",
      "keywords": null,
      "level": 1,
      "mainCategories": null,
      "media": null,
      "mediaId": null,
      "metaDescription": null,
      "metaTitle": null,
      "name": "Catalogue #1",
      "nestedProducts": null,
      "parent": null,
      "parentId": null,
      "parentVersionId": null,
      "path": null,
      "products": null,
      "seoUrls": null,
      "slotConfig": Array [],
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
        ],
        "customFields": Array [],
        "description": null,
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Catalogue #1",
        "slotConfig": Array [],
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "cc921c5fd5e2430ba878fb194c1aeb8a",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 2,
      "breadcrumb": Array [
        "Catalogue #1",
        "Clothing",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Iusto explicabo occaecati similique consequatur maxime dolores et reiciendis. Nulla similique assumenda suscipit commodi. Ut dolores et et. Recusandae quisquam dolores eaque placeat velit.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "cc921c5fd5e2430ba878fb194c1aeb8a",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "1d3da9a2f78f41cb9aeca34246f3da4c",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Clothing",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": Array [],
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Clothing",
        ],
        "customFields": Array [],
        "description": "Iusto explicabo occaecati similique consequatur maxime dolores et reiciendis. Nulla similique assumenda suscipit commodi. Ut dolores et et. Recusandae quisquam dolores eaque placeat velit.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Clothing",
        "slotConfig": Array [],
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "3a64e872ca404522a2c5d43ebc751e6b",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 3,
      "breadcrumb": Array [
        "Catalogue #1",
        "Accessories",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Nihil officiis ut enim reiciendis et eos expedita. Illo distinctio fugiat sed qui expedita delectus dolores. Sunt sunt laborum officia officiis qui fuga.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "3a64e872ca404522a2c5d43ebc751e6b",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "88ddcd14a4e84ee9b747776c7721643d",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Accessories",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Accessories",
        ],
        "customFields": Array [],
        "description": "Nihil officiis ut enim reiciendis et eos expedita. Illo distinctio fugiat sed qui expedita delectus dolores. Sunt sunt laborum officia officiis qui fuga.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Accessories",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "aeec8bca9dde431fb4cf822c291289bb",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 4,
      "breadcrumb": Array [
        "Catalogue #1",
        "Books",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "1484c88ca38c4d9caefd934265951afc",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Voluptates non expedita nobis occaecati est. Voluptas aliquid molestiae doloremque. Inventore et et iste dolorum.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "aeec8bca9dde431fb4cf822c291289bb",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "fbc30e021c9b4467a98c6421fc5b27ae",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Books",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Books",
        ],
        "customFields": Array [],
        "description": "Voluptates non expedita nobis occaecati est. Voluptas aliquid molestiae doloremque. Inventore et et iste dolorum.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Books",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "34b081d3d1044ab594cd522f672a929d",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 5,
      "breadcrumb": Array [
        "Catalogue #1",
        "Sports",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Est soluta distinctio error asperiores. Doloremque aut sit ipsa corporis exercitationem rerum maxime. Aut aut ut maxime sit ipsam. Eligendi illum eligendi aperiam.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "34b081d3d1044ab594cd522f672a929d",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "3fe64d91c1e44d7bb170feba608974a7",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Sports",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Sports",
        ],
        "customFields": Array [],
        "description": "Est soluta distinctio error asperiores. Doloremque aut sit ipsa corporis exercitationem rerum maxime. Aut aut ut maxime sit ipsam. Eligendi illum eligendi aperiam.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Sports",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "eea6df1898c940769d6b8d578be785a1",
      "active": false,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 6,
      "breadcrumb": Array [
        "Catalogue #1",
        "Books, Outdoors & Automotive",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Blanditiis reprehenderit quasi eos quos. Quia velit repellendus quaerat repellat voluptatem velit ut. Quod omnis quia porro fuga beatae sunt sunt.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "eea6df1898c940769d6b8d578be785a1",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "10ce0edc5085470bb6c1098a68a4f37f",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Books, Outdoors & Automotive",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Books, Outdoors & Automotive",
        ],
        "customFields": Array [],
        "description": "Blanditiis reprehenderit quasi eos quos. Quia velit repellendus quaerat repellat voluptatem velit ut. Quod omnis quia porro fuga beatae sunt sunt.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Books, Outdoors & Automotive",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "83e555dee2d2425eba42b7c628b65c67",
      "active": false,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 7,
      "breadcrumb": Array [
        "Catalogue #1",
        "Music",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Officiis rem ducimus quod atque rerum ducimus inventore. Vel veritatis et et culpa. Debitis nisi qui dolorem illo voluptatum reprehenderit porro.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "83e555dee2d2425eba42b7c628b65c67",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "34077a214f244f4d8bd1730f14964306",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Music",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Music",
        ],
        "customFields": Array [],
        "description": "Officiis rem ducimus quod atque rerum ducimus inventore. Vel veritatis et et culpa. Debitis nisi qui dolorem illo voluptatum reprehenderit porro.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Music",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "c0ad292c72c84b7782ecdd8c7d3c73bc",
      "active": false,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 8,
      "breadcrumb": Array [
        "Catalogue #1",
        "Kids & Toys",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Perspiciatis debitis quo iusto autem sapiente. Facilis consequatur nisi et. Soluta sed qui sapiente sit voluptatem rem. Aut fugiat dolorum quasi velit qui.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "c0ad292c72c84b7782ecdd8c7d3c73bc",
      "keywords": null,
      "level": 2,
      "mainCategories": null,
      "media": null,
      "mediaId": "6868d1cc071945419f715fb8f1c4d098",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Kids & Toys",
      "nestedProducts": null,
      "parent": null,
      "parentId": "3f637f17cd9f4891a2d7625d19fb37c9",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Kids & Toys",
        ],
        "customFields": Array [],
        "description": "Perspiciatis debitis quo iusto autem sapiente. Facilis consequatur nisi et. Soluta sed qui sapiente sit voluptatem rem. Aut fugiat dolorum quasi velit qui.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Kids & Toys",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "e8ff74bc4e4346208ea8a1a0a2312ac2",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 9,
      "breadcrumb": Array [
        "Catalogue #1",
        "Clothing",
        "Shoes & Clothing",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Autem quos porro eligendi placeat sit totam iusto. Sint voluptas hic modi atque. Omnis ea amet ex aut nulla corrupti. Est dolore aperiam ut eaque.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "e8ff74bc4e4346208ea8a1a0a2312ac2",
      "keywords": null,
      "level": 3,
      "mainCategories": null,
      "media": null,
      "mediaId": "10ce0edc5085470bb6c1098a68a4f37f",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Shoes & Clothing",
      "nestedProducts": null,
      "parent": null,
      "parentId": "cc921c5fd5e2430ba878fb194c1aeb8a",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|cc921c5fd5e2430ba878fb194c1aeb8a|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Clothing",
          "Shoes & Clothing",
        ],
        "customFields": Array [],
        "description": "Autem quos porro eligendi placeat sit totam iusto. Sint voluptas hic modi atque. Omnis ea amet ex aut nulla corrupti. Est dolore aperiam ut eaque.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Shoes & Clothing",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
    Object {
      "_uniqueIdentifier": "9429db6b2da348ea8308f1dbe6eee9f0",
      "active": true,
      "afterCategoryId": "mockedValue",
      "afterCategoryVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "autoIncrement": 10,
      "breadcrumb": Array [
        "Catalogue #1",
        "Clothing",
        "Computers",
      ],
      "childCount": "mockedValue",
      "children": null,
      "cmsPage": null,
      "cmsPageId": "7e360fe42925426b83563f9f135b1ff0",
      "createdAt": "2019-10-08T09:42:15+00:00",
      "customFields": null,
      "description": "Est debitis qui ipsam doloremque rerum voluptate. Et aliquam ea consectetur. Qui animi inventore officiis. Omnis aut optio magni cum.",
      "displayNestedProducts": true,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": "mockedValue",
          "versionId": null,
        },
      },
      "externalLink": null,
      "id": "9429db6b2da348ea8308f1dbe6eee9f0",
      "keywords": null,
      "level": 3,
      "mainCategories": null,
      "media": null,
      "mediaId": "15453518de284efda4693d141476ff79",
      "metaDescription": null,
      "metaTitle": null,
      "name": "Computers",
      "nestedProducts": null,
      "parent": null,
      "parentId": "cc921c5fd5e2430ba878fb194c1aeb8a",
      "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "path": "|3f637f17cd9f4891a2d7625d19fb37c9|cc921c5fd5e2430ba878fb194c1aeb8a|",
      "products": null,
      "seoUrls": null,
      "slotConfig": null,
      "tags": null,
      "translated": Object {
        "breadcrumb": Array [
          "Catalogue #1",
          "Clothing",
          "Computers",
        ],
        "customFields": Array [],
        "description": "Est debitis qui ipsam doloremque rerum voluptate. Et aliquam ea consectetur. Qui animi inventore officiis. Omnis aut optio magni cum.",
        "externalLink": null,
        "keywords": null,
        "metaDescription": null,
        "metaTitle": null,
        "name": "Computers",
        "slotConfig": null,
      },
      "translations": null,
      "type": "page",
      "updatedAt": "mockedValue",
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "visible": true,
    },
  ],
  "total": "mockedValue",
}
`;

exports[`shopware-6-client - E2E - CategoryService - getCategories should returns error when cannot find categories with given criteria 1`] = `[Error: Request failed with status code 500]`;
